src/puppeteer_pool.js

```diff
@@ -290,27 +318,26 @@ class PuppeteerPool {
     _killRetiredInstances() {
         log.debug('PuppeteerPool: retired browsers count', { count: _.values(this.retiredInstances).length });
 
-        _.mapObject(this.retiredInstances, (instance) => {
+        _.mapObject(this.retiredInstances, async (instance) => {
```

Comment:
I don't think this will work, `_.mapObject` will just swallow the promises and they might fail in the background

Reply:
It's the same as previous implementation, there were also promises only handled inside the loop. But there's a mistake on line 329, that line should also be in the try/catch. I'll fix that.

Reply:
okok